src/ucp/api/ucp.h

```diff
@@ -1015,6 +1028,32 @@ typedef struct ucp_params {
 } ucp_params_t;
 
 
+/**
+ * @ingroup UCP_CONTEXT
+ * @brief Lib attributes.
+ *
+ * The structure defines the attributes which characterize
+ * the Library.
+ */
+typedef struct ucp_lib_attr {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref ucp_lib_attr_field.
+     * Fields not specified in this mask will be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t          field_mask;
+
+    /**
+     * Max thread support of the library - permanent throughout the lifetime of
```

Comment:
Maximum level of thread support of the library, which is permanent throughout the lifetime of the library